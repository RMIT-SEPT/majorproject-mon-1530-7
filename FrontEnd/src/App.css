.App {
  text-align: center;
  background-color: rgba(164, 214, 197, 0.904);
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.page-container{
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.content-wrap{
  flex:1;
}