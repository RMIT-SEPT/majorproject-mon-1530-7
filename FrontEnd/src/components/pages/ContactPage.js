import React, { Component } from 'react'
import 'bootstrap/dist/css/bootstrap.css';
import { Container, Jumbotron, Button, Form} from 'react-bootstrap';


class ContactPage extends Component {
    render() {
        return (
            <Jumbotron id="jumbotron-custom">
                <Container>
                    <h1>
                        Contact Us.
                    </h1>    
                    <p>
                        We look forward to hearing what our cutz-omers have to say about our services, 
                        so please contact us and let us know how your experience with Fresh-Cutz was.
                    </p>

                    <p>
                        Phone Number : 1300-CUTZ
                        Email : FreshCutz@gmail.com
                    </p>

                    <h4>
                        OR Leave Feedback in this form below!
                    </h4>

                    <Form>
                        <Form.Group controlId="formName">
                            <Form.Label>Name</Form.Label>
                            <Form.Control type="text" placeholder="Enter Name"/>
                        </Form.Group>
                        <Form.Group controlId="formEmail">
                            <Form.Label>Email</Form.Label>
                            <Form.Control type="email" placeholder="Enter Email (If you want a reply!)" />
                        </Form.Group>
                        <Form.Group controlId="formFeedback">
                            <Form.Label>Feedback</Form.Label>
                            <Form.Control type="text" placeholder="Enter Feedback" />
                        </Form.Group>
                        <Button variant="primary" type="submit">
                            Leave Feedback!
                        </Button>
                    </Form>
                </Container>
            </Jumbotron>  
        );
    }
}

export default ContactPage;
