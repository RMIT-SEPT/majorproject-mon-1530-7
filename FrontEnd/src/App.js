import React from "react";
import { Redirect } from 'react-router';
import NavigationBar from "./components/layouts/Navbar.js";
import Footer from "./components/layouts/Footer.js";
import { Switch, Route, BrowserRouter } from "react-router-dom";
import LoginPage from "./components/pages/LoginPage";
import HomePage from "./components/pages/HomePage";
import AboutPage from "./components/pages/AboutPage";
import ContactPage from "./components/pages/ContactPage";
import CustomerBookingPage from "./components/pages/CustomerBookingPage.js";
import NewEmployeePage from "./components/pages/NewEmployeePage.js";

import AdminPastBooking from "./components/pages/AdminPastBooking.js";
import CustomerPastBookingsPage from "./components/pages/CustomerPastBookingsPage.js";

import Dashboard from "./components/pages/Dashboard.js";
import AdminDashboard from "./components/pages/AdminDashboard.js";
import Account from "./components/pages/Account.js";
import Profile from "./components/pages/Profile.js";

import ManageEmp from "./components/pages/ManageEmp.js";
import EmpDetails from "./components/pages/EmpDetails.js";

import UserProfile from "./UserProfile.js"
import AdminPastBookingsPage from "./components/pages/AdminPastBookingsPage.js";

function App() {
  console.log(UserProfile.getLoggedIn())
  if(UserProfile.getLoggedIn()) {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
                <Route exact path="/" component={HomePage} />,
                <Redirect from="/login" to="dashboard" />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Route path="/dashboard" component={Dashboard} >
                  {UserProfile.getAdmin() ? <Redirect to="/admin-dashboard" /> : <Dashboard />}
                </Route>,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Route path="/account" component={Account} >
                  {UserProfile.getLoggedIn() ? <Account /> : <Redirect to="/login"/>}
                </Route>,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} /> 
          
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
  else {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
              
                <Route exact path="/" component={HomePage} />,
                <Route path="/login" component={LoginPage} />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Redirect from="/dashboard" to="/login" />,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Redirect from="/account" to="/login" />,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} />  
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
}
 

export default App;
