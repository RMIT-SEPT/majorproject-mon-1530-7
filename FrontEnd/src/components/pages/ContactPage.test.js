import React from 'react';
import {MemoryRouter} from 'react-router';
import ContactPage from './ContactPage';
import App from '../../App';
import {mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing ContactPage route', () => {
    it('Should show the Contact Page without errors', () => {
      const component = mount( <MemoryRouter initialEntries={["/contact"]}>
          <App/>
       </MemoryRouter>
      );
      expect(component.contains(ContactPage));
    })
  })
