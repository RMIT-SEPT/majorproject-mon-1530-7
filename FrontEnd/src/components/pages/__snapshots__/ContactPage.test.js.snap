// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ContactPage route renders correctly 1`] = `
<Jumbotron
  fluid={false}
  id="jumbotron-alt2"
>
  <Container
    fluid={false}
  >
    <h2
      className="h2-center"
    >
      Contact Us
    </h2>
    <h5
      className="h5-center"
    >
      <img
        alt="icon"
        height="40"
        src="icon_5.png"
      />
      Phone Number : 1300 0000
    </h5>
    <h5
      className="h5-center"
    >
      <img
        alt="icon"
        height="40"
        src="icon_4.png"
      />
      Email : bookingsystem@gmail.com
    </h5>
    <Row
      noGutters={false}
    >
      <Col
        className="shadow p-3 mb-5 bg-white rounded"
        id="contact-card"
      >
        <h4
          className="h5-main3"
        >
          Have any questions?
        </h4>
        <Form
          inline={false}
        >
          <FormGroup
            controlId="formName"
            id="form-custom"
          >
            <FormControl
              placeholder="Enter Name"
              type="text"
            />
          </FormGroup>
          <FormGroup
            controlId="formEmail"
            id="form-custom"
          >
            <FormControl
              placeholder="Enter Email"
              type="email"
            />
          </FormGroup>
          <FormGroup
            controlId="formFeedback"
            id="form-custom"
          >
            <FormControl
              placeholder="Enter Feedback"
              type="text"
            />
          </FormGroup>
          <Link
            to="/signup"
          >
            <button
              className="btn-filled"
              id="align-btn"
            >
              Submit Now
            </button>
          </Link>
        </Form>
      </Col>
    </Row>
  </Container>
</Jumbotron>
`;
